### PR TITLE
Update Membership.md with more roles/details

### PR DESCRIPTION
**Pull request by karenhchu**
Signed-off-by: Karen Chu <karen.chu@microsoft.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: docs/MEMBERSHIP.md [hunk 1]
```diff
@@ -6,67 +6,167 @@
...
+  * Reviewed or merged at least 10 substantial PRs to the codebase
+  * Knowledgeable about the codebase
+  * Sponsored by a subproject approver
+    * With no objections from other approvers
+    * Done through PR to update the OWNERS file
+  * May either self-nominate, be nominated by an approver in this subproject
+
+* Privileges:
+  * Invitations to Contributor events
+  * Eligible to become a reviewer or approver
```
Comment (aramase) [inline docs/MEMBERSHIP.md:79]: This is in the `Reviewer` section. Should we update this to `Eligible to become an approver`?

Comment (aramase) [reply docs/MEMBERSHIP.md:79]: May lgtm code contributions

File: docs/MEMBERSHIP.md [hunk 2]
```diff
@@ -6,67 +6,167 @@
...
+  * Is supportive of new and infrequent contributors, and helps get useful PRs in shape to commit
+  * Mentoring new Maintainers
+  * Writing refactoring PRs
+  * Participating in CNCF Maintainer activities
+  * Determining strategy and policy for the project
+  * Participating in, and leading, community meetings
+* Requirements
+  * Experience as an Approver for at least 6 months   
+  * Demonstrates a broad knowledge of the project across multiple areas
+  * Is able to exercise judgement for the good of the project, independant of their employer, social circles, or teams
```
Comment (aramase) [inline docs/MEMBERSHIP.md:126]:
```suggestion
  * Is able to exercise judgement for the good of the project, independent of their employer, social circles, or teams
```

**Review by aramase — COMMENTED** [review]
@karenhchu Thank you for the PR!

Just 2 small nits. PTAL when you get a chance!

File: docs/MEMBERSHIP.md (hunk 1 above)
Comment (karenhchu) [inline docs/MEMBERSHIP.md:79]:
```suggestion
  * Eligible to become an approver
  * May give "lgtm" comment for code contributions
```

**Comment by aramase** [comment]
The switch to workload-identity is causing some failure with the gcp test. @tam7t is debugging the issue in this [PR](https://github.com/kubernetes-sigs/secrets-store-csi-driver/pull/635).

**Review by aramase — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**